discovery-provider/src/tasks/index_metrics.py

```diff
@@ -126,7 +136,154 @@ def refresh_metrics_matviews(db):
         session.execute('REFRESH MATERIALIZED VIEW app_name_metrics_all_time')
         logger.info('index_metrics.py | refreshed metrics matviews')
 
-######## CELERY TASK ########
+"""
+get number of discovery nodes and then the service info, including the endpoint, for each node
+return all node endpoints except that of this node
+"""
+def get_all_other_nodes():
+    shared_config = aggregate_metrics.shared_config
+    eth_web3 = aggregate_metrics.eth_web3
+    eth_registry_address = aggregate_metrics.eth_web3.toChecksumAddress(
+        shared_config["eth_contracts"]["registry"]
+    )
+    eth_registry_instance = eth_web3.eth.contract(
+        address=eth_registry_address, abi=eth_abi_values["Registry"]["abi"]
+    )
+    sp_factory_address = eth_registry_instance.functions.getContract(
+        sp_factory_registry_key
+    ).call()
+    sp_factory_inst = eth_web3.eth.contract(
+        address=sp_factory_address, abi=eth_abi_values["ServiceProviderFactory"]["abi"]
+    )
+    num_discovery_providers = sp_factory_inst.functions.getTotalServiceTypeProviders(discovery_node_service_type).call()
+    logger.info(f"number of discovery providers: {num_discovery_providers}")
+    service_infos = [sp_factory_inst.functions.getServiceEndpointInfo(discovery_node_service_type, i).call() \
+        for i in range(1, num_discovery_providers + 1)]
```

Comment:
just an optimization, don't need to add this, but if performance is/was an issue we can use threadpoolexecutors to parallelize this like https://github.com/AudiusProject/audius-protocol/blob/master/discovery-provider/src/tasks/index_network_peers.py#L66-L85

Reply:
i think doing this sync is probably better

Reply:
I like it!